storage/innobase/fil/fil0fil.cc

```diff
@@ -9278,6 +9283,8 @@ possibly cached by the OS.
 @param[in]	purpose		FIL_TYPE_TABLESPACE or FIL_TYPE_LOG, can be
 ORred. */
 void fil_flush_file_spaces(uint8_t purpose) {
+  if (!fil_system) return;
```

Comment:
when will we reach here without the fil_system initialization?

Reply:
We call fil_flush_file_spaces() from srv_sgutdown(). Now srv_sgutdown() work and w/ error inititialization, but w/ error inititialization may be w/o the fil_system initialization

Reply:
I see that fil_init() is called at very early stage of srv_start(). This means fil_system should be initialized. Is it possible that we are erroring out without srv_start() being called?

Can you please make it as an assert ut_a(fil_system != nullptr) and get stack trace?

Reply:
See https://github.com/tempesta-tech/percona-server/blob/avb-log_file_name/storage/innobase/srv/srv0start.cc#L2192
This return before fil_init()